pytest_tests/testsuites/conftest.py

```diff
@@ -304,33 +305,21 @@ def background_grpc_load(client_shell: Shell, hosting: Hosting):
                 k6_verify_instance.wait_until_finished(BACKGROUND_LOAD_MAX_TIME)
 
 
-@pytest.fixture(scope="session")
-@allure.title("Prepare wallet and deposit")
-def default_wallet(client_shell: Shell, temp_directory: str, cluster: Cluster):
-    wallet_path = os.path.join(os.getcwd(), ASSETS_DIR, f"{str(uuid.uuid4())}.json")
-    init_wallet(wallet_path, WALLET_PASS)
-    allure.attach.file(wallet_path, os.path.basename(wallet_path), allure.attachment_type.JSON)
-
-    if not FREE_STORAGE:
-        main_chain = cluster.main_chain_nodes[0]
-        deposit = 30
-        transfer_gas(
-            shell=client_shell,
-            amount=deposit + 1,
-            main_chain=main_chain,
-            wallet_to_path=wallet_path,
-            wallet_to_password=WALLET_PASS,
-        )
-        deposit_gas(
-            shell=client_shell,
-            main_chain=main_chain,
-            amount=deposit,
-            wallet_from_path=wallet_path,
-            wallet_from_password=WALLET_PASS,
-        )
+@pytest.fixture(scope="function")
+@allure.title("Prepare not owner wallet and deposit")
+def not_owner_wallet(client_shell: Shell, temp_directory: str, cluster: Cluster):
+    wallet_path = create_wallet(client_shell, temp_directory, cluster, 'not_owner_wallet')
+    try:
+        yield wallet_path
+    finally:
```

Comment:
as far as I remember, there is no need to handle exceptions in fixtures like these, in case of errors inside other fixtures/tests - the teardown code will still be invoked https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#handling-errors-for-yield-fixture

Reply:
> as far as I remember, there is no need to handle exceptions in fixtures like this, in case of errors inside other fixtures/tests - the teardown code will still be invoked https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#handling-errors-for-yield-fixture

Fixed